src/signal/signal.c

```diff
@@ -0,0 +1,38 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   signal.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: lporoshi <lporoshi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/12/17 14:10:06 by lporoshi          #+#    #+#             */
+/*   Updated: 2023/12/17 14:22:17 by lporoshi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <readline/readline.h>
+#include <readline/history.h>
+#include "libft.h"
+
+void	sigint_handler(int sig)
+{
+	if (sig == SIGINT)
+	{
+		rl_redisplay();
+		write(1, "\n", 1);
+		rl_on_new_line();
+		rl_replace_line("", 0);
+		rl_redisplay();
+	}
+	return ;
+}
+
+void	init_signal_handlers(void)
```

Comment:
does ctrl+d work without handler or it's just not implemented yet?

Reply:
Pressing Ctrl-D doesn't send a linux signal, but a special EOF character with weird behavior, so it should be handled separately from things like Ctrl-C.